archive-static-sites/x-archive/src/components/TweetsDisplayComponent.vue

```diff
@@ -0,0 +1,42 @@
+<script setup lang="ts">
+import { defineProps, ref, computed } from 'vue'
+
+import { Tweet } from '../types'
+import TweetComponent from '../components/TweetComponent.vue'
+
+const props = defineProps<{
+    tweets: Tweet[];
+    tweetTypePlural: string;
+}>();
+
+const filterText = ref('');
+
+const filterTextPlaceholder = computed(() => {
+    return `Filter your ${props.tweetTypePlural}`;
+});
+
+const filteredTweets = computed(() => {
+    return props.tweets.filter(tweet => {
+        const lowerCaseFilterText = filterText.value.toLowerCase();
+        const tweetText = tweet.text ? tweet.text.toLowerCase() : '';
+        const tweetUsername = tweet.username ? tweet.username.toLowerCase() : '';
+        return tweetText.includes(lowerCaseFilterText) || tweetUsername.includes(lowerCaseFilterText);
+    });
+});
+</script>
+
+<template>
+    <div class="tweets-container">
+        <div class="filter-container">
+            <p><input type="text" v-model="filterText" class="form-control" :placeholder="filterTextPlaceholder"></p>
```

Comment:
Probably needs to be addressed in a separate accessibility audit of the entire app, but inputs should be associated with a label. Placeholders are not good labels.